pxtlib/emitter/emitter.ts

```diff
@@ -321,6 +321,9 @@ namespace ts.pxtc {
         paramHelp?: pxt.Map<string>;
         // foo.defl=12 -> paramDefl: { foo: "12" }
         paramDefl: pxt.Map<string>;
+
+        min?: Map<string>; // min range
```

Comment:
I store the minimum and maximum values by name using a Map because when I create the shadow value I don't have the explicit position of the parameter in the parameters array of the block.

![image](https://cloud.githubusercontent.com/assets/25017756/23439752/136d0ddc-fdcd-11e6-9a99-40d3f12969ab.png)